test/test_tensordict.py

```diff
@@ -1438,6 +1438,37 @@ def test_getitem_batch_size():
         assert expected_shape == resulting_shape, idx
 
 
+@pytest.mark.parametrize("device", get_available_devices())
+def test_requires_grad(device):
```

Comment:
This looks great to test that tensordict support gradients when created, but what about modification of existing tensordicts?


Do you think it would be possible to also test this with `TestTensorDicts`? 
You can go through various examples of TensorDicts with 
```python
    @pytest.mark.parametrize(
        "td_name",
        [
            "td",
            "stacked_td",
            "sub_td",
            "idx_td",
            "saved_td",
            "unsqueezed_td",
            "td_reset_bs",
        ],
    )
    def test_blabla(td_name):
        exec()
```

What you would need to do is replace one tensor of the tensordict with a tensor that requires a gradient and check that this is supported. Something like
```python
td.set("a", torch.randn_like(td.get("a")).requires_grad_())
```
You'd need to check that the MetaTensor associated with that tensor is updated too.
```python
assert td.get_meta("a").requires_grad
```